polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1298,7 +1302,12 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
         validateLocationsForTableLike(tableIdentifier, dataLocations, resolvedStorageEntity);
         // also validate that the table location doesn't overlap an existing table
         dataLocations.forEach(
-            location -> validateNoLocationOverlap(tableIdentifier, resolvedNamespace, location));
+            location ->
+                validateNoLocationOverlap(
+                    CatalogEntity.of(resolvedStorageEntity.getRawFullPath().getFirst()),
```

Comment:
Looks like we kind of have a lot of vestigial re-parsing of catalogPath `getFirst()` scattered around for getting the CatalogEntity, but the `private final CatalogEntity catalogEntity` of the `BasePolarisCatalog` should really be the authoritative source of catalog properties.

`getFirst()` is kind of risky because in some contexts (mostly in `PolarisAdminService`) the resolution path includes the "RootContainerEntity" that represents the entire realm's root as the first element of a resolution path. There isn't really supposed to be any assumptions about what the type of the first element of a resolution path is.

Since this is also used in other "validate location" methods I'm fine waiting to clean up all these `getFirst()` calls in a separate refactor PR, so leave it up to you whether you prefer to go ahead and use `catalogEntity` now or leave it to a bulk refactor.

Reply:
Good callout, I will replace this where possible. I won't necessarily hunt down the existing usages though, so we should still go back and refactor at some point.